models/articles_models.js

```diff
@@ -35,4 +35,14 @@ function checkArticleIDExists(article_id){
     })
 }
 
-module.exports = { fetchArticleById, fetchAllArticles,checkArticleIDExists }
\ No newline at end of file
+function updateArticleById(article_id, inc_votes){
+    return Promise.all([db.query(`UPDATE articles SET votes = votes + $1 WHERE article_id = $2 RETURNING *`, [inc_votes,article_id]),checkArticleIDExists(article_id)])
+    .then(([{rows}]) => {
+        const updatedArticle = rows[0];
+        if(updatedArticle.votes < 0){
+            updatedArticle.votes = 0
```

Comment:
Im not sure we need a promise.all or to use checkArticleId exists. The update should give us 0 rows if there is an error which we can catch :)